src/cargo/core/package_id_spec.rs

```diff
@@ -170,7 +176,8 @@ impl PackageIdSpec {
         }
 
         if let Some(ref v) = self.version {
-            if v != package_id.version() {
+            let req = v.exact_req();
+            if !req.matches(package_id.version()) {
```

Comment:
This causes us to no longer require the `build` field matches which causes problems.  Working on a fix which we'll need to backport.

Reply:
See #12806